From 67bc21522700ce9bde088364b7553782d015edbc Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@nabladev.com>
Date: Fri, 31 Oct 2025 04:52:10 +0100
Subject: [PATCH] ARM: stm32: Add missing build of ST DFU virt code on DH
 STM32MP1 DHSOM

Commit 6d91f0a3a14d ("board: st: common: cleanup dfu support") split
the vendor-specific DFU implementation into two files, but failed to
update other non-ST boards. This did not lead to noticable breakage
with plain simple dfu-util, but it makes the ST proprietary programmer
CLI tool end in an infinite loop. Update the Makefile accordingly to
allow even that kind of tooling to work.

Upstream-Status: Submitted [https://lore.kernel.org/u-boot/20251031041640.174689-1-marek.vasut@mailbox.org/]
Fixes: 6d91f0a3a14d ("board: st: common: cleanup dfu support")
Signed-off-by: Marek Vasut <marex@nabladev.com>
---
 board/dhelectronics/dh_stm32mp1/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/board/dhelectronics/dh_stm32mp1/Makefile b/board/dhelectronics/dh_stm32mp1/Makefile
index ffd221ac224..2f4a301d1a0 100644
--- a/board/dhelectronics/dh_stm32mp1/Makefile
+++ b/board/dhelectronics/dh_stm32mp1/Makefile
@@ -7,3 +7,4 @@ obj-$(CONFIG_PMIC_STPMIC1) += ../../st/common/stpmic1.o
 obj-y += board.o
 
 obj-$(CONFIG_SET_DFU_ALT_INFO) += ../../st/common/stm32mp_dfu.o
+obj-$(CONFIG_$(PHASE_)DFU_VIRT) += ../../st/common/stm32mp_dfu_virt.o
-- 
2.51.0

